File: kumoh-code-2017/Java/#1 기본/#2 수업/소스코드/10. 은행관리 프로그램 (ppt)/은행관리/Transaction.java
package 은행관리;

abstract public class Transaction
{
	private int account_number; // Model's Account
	private Screen screen; // UI Component
	private BankDatabase bank_database; // UI Component
	
	public Transaction(int account_number, Screen screen, BankDatabase bank_database)
	{
		this.account_number = account_number;
		this.screen = screen;
		this.bank_database = bank_database;
	}
	
	public int getAccountNumber(){return account_number;}
	public Screen getScreen(){return screen;}
	public BankDatabase getBankDatabase(){return bank_database;}
	
	abstract public void execute(); // Transaction 실행
}

File: kumoh-code-2017/C++/#2 설계/과제15 은행관리 프로그램/Transaction.cpp
#include"Transaction.h"

Transaction::Transaction(){
	accountNumber = 0;
}

Transaction::Transaction(int accountNumber, Screen screen, BankDatabase bankDB){
	this->accountNumber = accountNumber;
	this->screen = screen;
	this->bankDB = bankDB;
}

int Transaction::getAccountNumber(){
	return accountNumber;
}

Screen Transaction::getScreen(){
	return screen;
}

BankDatabase Transaction::getBankDataBalance(){
	return bankDB;
}

